### ISSUE TITLE
ASB 2017.12

### ISSUE THREAD

**Issue by MSe1969** [cm12-amami/discussion#9]
@hahnjo @Micha-Btz 
Google security patches are out, no AOSP links, yet

**Comment by MSe1969** [comment]
**1st analysis on the 2017-12-05 patch level:**

*Kernel components*

- [x] CVE-2017-13162 	A-64216036* 	[N/A - patch]
- [ ] CVE-2017-0564 	A-34276203* 	[N/A - not publicly avialable]
- [x] CVE-2017-7533 	A-63689921	potentially applicable/backport required
- [ ] CVE-2017-13174 	A-63100473* 	[N/A - not publicly avialable] - leaving out, doesn't compile

*MediaTek components*
CVE-2017-13170 	A-36102397*	[N/A - not publicly avialable]
CVE-2017-13171 	A-64316572*	[N/A - not publicly avialable]
CVE-2017-13173 	A-28067350*	[N/A - not publicly avialable]

*NVIDIA components*
CVE-2017-6262 	A-38045794*	[N/A - not publicly avialable]
CVE-2017-6263 	A-38046353*	[N/A - not publicly avialable]
CVE-2017-6276 	A-63802421*	[N/A - not publicly avialable]

*Qualcomm components*
CVE-2017-11043 	A-64728953	[N/A - file(s) do(es) not exist]
CVE-2016-3706 	A-34499281	[N/A - file(s) do(es) not exist]
CVE-2016-4429 	A-68946906	[N/A - file(s) do(es) not exist]
CVE-2017-11007 	A-66913719	[N/A - file(s) do(es) not exist]
CVE-2017-14904 	A-63662821*	[N/A - not publicly avialable]
CVE-2017-9716 	A-63868627	[N/A - file(s) do(es) not exist]
- [ ] CVE-2017-14897 	A-65468973	potentially applicable/backport required
    => does not fit our code
CVE-2017-14902 	A-65468970	[N/A - file(s) do(es) not exist]
- [ ] CVE-2017-14895 	A-65468977	potentially applicable/backport required
    => does not fit our code
*Qualcomm closed-source components*
... as the header already indicates...

**Comment by Micha-Btz** [comment]
2017-12-01 security patch level—Vulnerability details
Framework

CVE-2017-0807 | EoP | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2 => N/A (not publicly available)
- [x] CVE-2017-0870 | EoP | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0

Media framework

- [x] CVE-2017-0837 | EoP | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0



System

- [x] CVE-2017-13156 EoP | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0
- [x] CVE-2017-13157 ID | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0
- [x] CVE-2017-13158 ID | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0
- [x] CVE-2017-13159 ID | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0

I will start add patches today.

**Comment by MSe1969** [comment]
This time, there is a lot to think around backporting... :(  - really awful...
I am making some good progress, but this ASB isn't an easy one

**Comment by MSe1969** [comment]
Have uploaded the checked patches above to cm-12.1-test
CVE-2017-0807 is not publicly available, so can be removed from the list for now
CVE-2017-13156 is more complicated, but we DEFINITELY SHOULD implement it, see here:
https://www.guardsquare.com/en/blog/new-android-vulnerability-allows-attackers-modify-apps-without-affecting-their-signatures

I am really struggling with CVE-2017-13156 - maybe taking the libziparchive/ from N and get that to compile would be a better approach (tried already, but that has a side effect on getting a static host library in place...)? @hahnjo - maybe your C expertise could help to look deeper into the cm-12.1 sources and get what we want?

**Comment by Micha-Btz** [comment]
Pixel / Nexus Security Bulletin—December 2017 

CVE | References | Type | Severity | Updated AOSP versions
Media framework

- [x] CVE-2017-13154 | A-63666573 | EoP | High | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0
- [x] CVE-2017-0879 | A-65025028 |DoS | High | 5.1.1, 6.0, 6.0.1
- [ ] CVE-2017-13149 | A-65719872 DoS | High | 5.1.1, 6.0, 6.0.1 - doesn't fit our code
- [x] CVE-2017-13152 | A-62872384 | ID | Moderate | 5.1.1, 6.0, 6.0.1, 7.0, 7.1.1, 7.1.2, 8.0


Kernel components

- [x] CVE-2017-13167 | A-37240993* | EoP | High | Sound timer
- [x] CVE-2017-13163 | A-37429972* | EoP | Moderate | MTP USB driver
- [ ] CVE-2017-15868 | A-33982955         Upstream kernel | EoP | Moderate | Wireless driver Bluetooth -leaving this out, oesn't compile
- [ ] CVE-2017-13165 | A-31269937* | EoP | Moderate | File system
- [ ] CVE-2017-13166 | A-34624167* | EoP | Moderate | V4L2 video driver
- [x] CVE-2017-1000380 | A-64217740          Upstream kernel | EoP | Moderate | Sound timer driver
- [x] CVE-2017-13168 | A-65023233* | EoP | Moderate | SCSI driver
- [ ] CVE-2017-13169 | A-37512375* | ID | Moderate | Camera server
- [x] CVE-2017-13164 | A-36007193* | ID | Moderate | Binder driver

I have not found a patch to all kernel cve.

Qualcomm components
CVE | References | Type | Severity | Component
-- | -- | -- | -- | --
CVE-2017-9708 | A-62674846*         QC-CR#2081806 | EoP | Moderate | Kernel
CVE-2017-11042 | A-38232268*         QC-CR#2070438 | EoP | Moderate | Binder
CVE-2017-11030 | A-64431967         QC-CR#2034255      [2] | EoP | Moderate | Display
CVE-2017-9703 | A-34329758*         QC-CR#2038086 | EoP | Moderate | Video driver
CVE-2017-9718 | A-36386076*         QC-CR#2045918 | EoP | Moderate | Kernel
CVE-2017-8244 | A-35138888*         QC-CR#2013361 | EoP | Moderate | Debugfs driver
CVE-2017-14901 | A-65468984          QC-CR#2059714 | EoP | Moderate | WLAN
CVE-2017-9698 | A-63868678          QC-CR#2023860 | EoP | Moderate | Graphics
CVE-2017-9700 | A-63868780          QC-CR#2043822 | EoP | Moderate | Audio
CVE-2017-9722 | A-64453224         QC-CR#2034239      [2] | EoP | Moderate | Display
CVE-2017-11049 | A-64728945          QC-CR#2034909 | EoP | Moderate | Display
CVE-2017-11047 | A-64728948          QC-CR#2057285 | EoP | Moderate | Display
CVE-2017-14898 | A-65468978          QC-CR#2054748 | EoP | Moderate | WLAN
CVE-2017-14899 | A-65468980          QC-CR#2054752 | EoP | Moderate | WLAN
CVE-2017-11044 | A-65468989          QC-CR#2063166 | EoP | Moderate | Graphics
CVE-2017-11045 | A-65468993          QC-CR#2060377  [2] | EoP | Moderate | Camera
CVE-2017-14900 | A-65468983          QC-CR#2058468 | EoP | Moderate | WLAN
CVE-2017-9710 | A-63868933          QC-CR#2023883 | EoP | Moderate | Data HLOS
CVE-2017-11019 | A-64453105          QC-CR#2030638 | EoP | Moderate | Display
CVE-2017-11016 | A-64453423          QC-CR#2038685 | EoP | Moderate | Audio
CVE-2017-11033 | A-64453422         QC-CR#2031930      [2] | EoP | Moderate | Kernel
CVE-2017-14896 | A-65468975          QC-CR#2013716 | EoP | Moderate | GUD mobicore driver
CVE-2017-8281 | A-62378232          QC-CR#2015892 | ID | Moderate | DCI driver
CVE-2017-14903 | A-63522505*         QC-CR#2088768 | ID | Moderate | WLAN
CVE-2017-11031 | A-64442463          QC-CR#2059181 | ID | Moderate | Display
CVE-2017-14905 | A-37719782          QC-CR#2061251 | ID | Moderate | Wireless networking

**Comment by MSe1969** [comment]
@Micha-Btz - is it correct that we have now everything, except for the Qualcomm list, which we still need to analyze in detail?

**Comment by Micha-Btz** [comment]
I think so but I have to revert some commits to get the kernel compile. Something is strange here will will create a new kernel branch and readd the patches to test who the problem really is.

**Comment by MSe1969** [comment]
How can I help testing? >> Should I simply use the cm-12.1-dec branch for testing?
Also: Are the qualcomm components of the Pixel/Nexus bulletin already part of the cm-12.1-dec repo (as far as they're applicable)? (Don't see checkboxes there...)

**Comment by Micha-Btz** [comment]
I have committed to the local manifest to use the dec kernel branch. So far the build run fine.
But I have not yet installed it on my phone.
I have found patches for binder CVE-2017-13162 & CVE-2017-13164 but in a quick test yesterday the don't apply cleanly.
Until know there are no patches for qualcomm added and I don't have take a look at these patches.
If you want to do it then go on. 
This month it is a really huge one. Sometimes android bothers me.

**Comment by MSe1969** [comment]
Here comes my status regarding the Pixel/Nexus Qualcomm components:

CVE-2017-9708 	A-62674846* QC-CR#2081806 Moderate 	not found*
CVE-2017-11042 	A-38232268* QC-CR#2070438 Moderate 	not found*
CVE-2017-11030 	A-64431967  QC-CR#2034255 Moderate 	N/A - code does not fit to ours
CVE-2017-9703 	A-34329758* QC-CR#2038086 Moderate 	N/A - found, but file not exists
CVE-2017-9718 	A-36386076* QC-CR#2045918 Moderate 	not found*
CVE-2017-8244 	A-35138888* QC-CR#2013361 Moderate 	**implemented**
CVE-2017-14901 	A-65468984  QC-CR#2059714 Moderate 	N/A - code does not fit to ours
CVE-2017-9698 	A-63868678  QC-CR#2023860 Moderate 	N/A - file does not exist (4.4)
CVE-2017-9700 	A-63868780  QC-CR#2043822 Moderate 	N/A - file does not exist (4.4)
CVE-2017-9722 	A-64453224  QC-CR#2034239[2] Moderate 	N/A - code does not fit to ours
CVE-2017-11049 	A-64728945  QC-CR#2034909 Moderate      N/A - file does not exist (3.18)
CVE-2017-11047 	A-64728948  QC-CR#2057285 Moderate 	N/A - file does not exist (4.4)
CVE-2017-14898 	A-65468978  QC-CR#2054748 Moderate	N/A - code does not fit to ours
CVE-2017-14899 	A-65468980  QC-CR#2054752 Moderate 	N/A - code does not fit to ours
CVE-2017-11044 	A-65468989  QC-CR#2063166 Moderate 	N/A - code does not fit to ours
CVE-2017-11045 	A-65468993  QC-CR#2060377[2] Moderate 	N/A - file does not exist (3.18)
CVE-2017-14900 	A-65468983  QC-CR#2058468 Moderate 	N/A - code does not fit to ours
CVE-2017-9710 	A-63868933  QC-CR#2023883 Moderate 	N/A - file does not exist (4.4)
CVE-2017-11019 	A-64453105  QC-CR#2030638 Moderate	N/A - code does not fit to ours
CVE-2017-11016 	A-64453423  QC-CR#2038685 Moderate 	N/A - file does not exist (3.18)
CVE-2017-11033 	A-64453422  QC-CR#2031930[2] Moderate 	**implemented**
CVE-2017-14896 	A-65468975  QC-CR#2013716 Moderate      *[NOT sure about this one???]*
CVE-2017-8281 	A-62378232  QC-CR#2015892 Moderate  	N/A - code does not fit to ours
CVE-2017-14903 	A-63522505* QC-CR#2088768 Moderate 	N/A - found, but does not fit our code
CVE-2017-11031 	A-64442463  QC-CR#2059181 Moderate 	N/A - file does not exist (4.4)
CVE-2017-14905 	A-37719782  QC-CR#2061251 Moderate 	N/A - already fixed

CVE-2017-14896 (the "not sure" one) aims at removing the entire GUD mobicore driver (trusted zone etc.) - the patch is in 3.18 and the commit message talks about a replacement - so if you ask me, we should not mess with that in our 3.4 kernel, this sounds like a recipe for trouble... 

So we should now have everything - and should test again...

**Comment by MSe1969** [comment]
The cm-12.1-dec kernel branch seems to run stable, also with my two additional commits.
Should we consider shipping? Is all complete? Or would you like to still do some tests?

**Comment by Micha-Btz** [comment]
It's not all complete the binder patches are missing but I will need some time to analyse why the patches are not apply cleanly. Maybe next week maybe after chrismas.
Best way for now I think is:
delete the kernel cm-12.1-test branch
create a new test branch from dec.
then ship the build
then add more patches and maybe ship it in between or with the January patches

**Comment by Micha-Btz** [comment]
It's not all complete the binder patches are missing but I will need some time to analyse why the patches are not apply cleanly. Maybe next week maybe after chrismas.
Best way for now I think is:
delete the kernel cm-12.1-test branch
create a new test branch from dec.
then ship the build
then add more patches and maybe ship it in between or with the January patches

**Comment by MSe1969** [comment]
Okay, I will take care accordingly

**Comment by MSe1969** [comment]
I see you have added four further commits to the dec branch today. Should I rebuild the test branch **without** the last four patches? (This is what I tested and what runs okay so far) - please advise

**Comment by Micha-Btz** [comment]
the build runs fine with the four new patches. I will install the ROM now and test. Maybe wait until tomorrow

**Comment by MSe1969** [comment]
okay, I'll wait until I read from you. What are those four patches? Did you find some more "not publicly available"? Should I also build and install on my phone?

**Comment by Micha-Btz** [comment]
the patches fix CVE-2017-0564.Why not build run it. But I think it works. I can write you from phone.

**Comment by MSe1969** [comment]
My device runs fine, including the four patches. I am writing you from the device. So I suggest I'll ship tonight and merge. Waiting for your feedback

**Comment by Micha-Btz** [comment]
do it.I have also no problems

Am 14. Dezember 2017 17:22:29 MEZ schrieb MSe1969 <notifications@github.com>:
>My device runs fine, including the four patches. I am writing you from
>the device. So I suggest I'll ship tonight and merge. Waiting for your
>feedback

**Comment by MSe1969** [comment]
Okay, shipped and merged into stable branch.
I understand you would like to keep this issue open because of the binder patches?

**Comment by Micha-Btz** [comment]
Yes, let it open for now.

**Comment by Micha-Btz** [comment]
patches for binder pushed. testing now

**Comment by MSe1969** [comment]
Hi, 
I have backported two commits from cm-13.0 into build (test branch), to have the product file start with "lineage" instead of "cm", tested successfully. But I have not yet picked any commits changing the internal build structure. Those commits would be:
https://review.lineageos.org/#/c/154598/
https://review.lineageos.org/#/c/154599/
https://review.lineageos.org/#/c/154595/
Should I go ahead and work on them as well? The benefit would be that it would potentially ease to work on further device repo's to be used with our build tree.

Wrt the binder patches: 
Please let me know when I can either test on my end (i.e. runs "stable enough" to test) or to which specific issue I should have a deeper look to be able to help spotting an issue. If we all change coding in parallel, this would not make a lot of sense here.
Thanks M.

**Comment by Micha-Btz** [comment]
the binder patches are ready I think. The build I did yesterday runs fine. 

Am 16. Dezember 2017 15:43:54 MEZ schrieb MSe1969 <notifications@github.com>:
>Hi, 
>I have backported two commits from cm-13.0 into build (test branch), to
>have the product file start with "lineage" instead of "cm", tested
>successfully. But I have not yet picked any commits changing the
>internal build structure. Those commits would be:
>https://review.lineageos.org/#/c/154598/
>https://review.lineageos.org/#/c/154599/
>https://review.lineageos.org/#/c/154595/
>Should I go ahead and work on them as well? The benefit would be that
>it would potentially ease to work on further device repo's to be used
>with our build tree.
>
>Wrt the binder patches: 
>Please let me know when I can either test on my end (i.e. runs "stable
>enough" to test) or to which specific issue I should have a deeper look
>to be able to help spotting an issue. If we all change coding in
>parallel, this would not make a lot of sense here.
>Thanks M.

**Comment by Micha-Btz** [comment]
For me it is equal if it is called cm or lineage. Go ahead. I will merge the kernel dec branch to the test branch and then we can to as usual. Then I will close this issue since we now have all public available patches.

**Comment by Micha-Btz** [comment]
merged. so i close this.
